core/jwks/src/validator.ts

```
@@ -0,0 +1,283 @@
+export interface JWKSValidationResult {
+  isValid: boolean;
+  validKeysCount: number;
+  currentKeysCount: number;
+  totalKeysCount: number;
+  validationErrors: string[];
+  warningMessages: string[];
+  keyResults: KeyValidationResult[];
+}
+
+export class JWKSValidationError extends Error {
+  constructor(message: string, public readonly code: string, public readonly details?: any) {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)